ASP.NET API/WAVC_WebApi/Controllers/FriendsManagerController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WAVC_WebApi.Data;
+using WAVC_WebApi.Models;
+
+namespace WAVC_WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    [ApiController]
+    public class FriendsManagerController : ControllerBase
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ApplicationDbContext _dbContext;
+        private readonly FriendsManager _friendsManager;
+        private readonly int _searchResults = 10;
+
+        public FriendsManagerController(UserManager<ApplicationUser> userManager, ApplicationDbContext dbContext)
+        {
+            _userManager = userManager;
+            _dbContext = dbContext;
+            _friendsManager = new FriendsManager(dbContext);
+        }
+
+        [HttpGet]
+        [Route("Friends")]
+        public async Task<List<ApplicationUserModel>> GetFriends()
+        {
+            var sender = await _userManager.GetUserAsync(HttpContext.User);
+            if (sender != null)
+            {
+                return _friendsManager.GetFriends(sender).Select(u => u.ToApplicationUserModel()).ToList();
+            }
+            else
+            {
+                return new List<ApplicationUserModel>();
+            }
+        }
+
+        [HttpGet]
+        [Route("RequestsForUser")]
+        public async Task<List<ApplicationUserModel>> GetRequestsForUser()
+        {
+            var sender = await _userManager.GetUserAsync(HttpContext.User);
+            if (sender != null)
+            {
+                return _friendsManager.GetRequestsForUser(sender).Select(u => u.ToApplicationUserModel()).ToList();
+            }
+            else
+            {
+                return new List<ApplicationUserModel>();
+            }
+        }
+
+        [HttpGet]
+        [Route("CurrentFriend")]
+        public async Task<ApplicationUserModel> GetCurrentFriend(string name, string surname)
```

Comment:
This function should look like:
> GET /api/Friends/user_id

If you want to pass anything in body, create special model class for that.

Reply:
To comply with other routes i'd suggest `FriendId` instead. It makes it obvious that we'll only get id of a friend and other routes don't use underscore.